### PR TITLE
Build: Fix issue with test status logging

### PR DESCRIPTION
**Pull request by matriv**
Handle the case of `Description` being null which is a valid case as
described in the `HeartBeatEvent`'s javadoc, which previously resulted
in exceptions that "pollute" the build output.

Follows: #28563

### CODE REVIEW

File: buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy [hunk 1]
```diff
@@ -146,31 +147,45 @@ class TestProgressLogger implements AggregatedEventListener {
...
      * Extract a Class#getSimpleName style name from Class#getName style
      * string. We can't just use Class#getSimpleName because junit descriptions
      * don't always set the class field but they always set the className
      * field.
      */
-    private static String simpleName(String className) {
-        return className.substring(className.lastIndexOf('.') + 1)
+    private static String simpleName(Description description) {
+        if (description == null) {
+            return "<unknown>"
```
Comment (rjernst) [inline buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy]: Maybe this should indicate unknown class, and in the test name we should add unknown test?

Comment (matriv) [reply buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy]: My thinking is if the description is null then we cannot know neither.
I believe, there is no case that the class is unknown but the method isn't or the opposite.

Comment (rjernst) [reply buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy]: I mean that we should be more explicit with unknown class vs unknown test.

**Comment by matriv** [comment]
@rjernst thx, fixup pushed.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**